web/components/roadmap/MiniRoadmap.test.tsx

```diff
@@ -91,18 +95,77 @@ describe("<MiniRoadmap />", () => {
       ],
     });
 
-    useMockUserData({
+    const subject = renderMiniRoadMap(
+      "task3",
+      generateUserData({
+        preferences: generatePreferences({
+          roadmapOpenSections: ["PLAN", "START"],
+        }),
+      })
+    );
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  describe("User Step State Preferences", () => {
+    const userData = generateUserData({
       preferences: generatePreferences({
         roadmapOpenSections: ["PLAN", "START"],
+        roadmapOpenSteps: [2],
       }),
     });
+    beforeEach(() => {
+      useMockRoadmap({
+        steps: [
+          generateStep({
+            name: "step1",
+            step_number: 1,
+            section: "PLAN",
+            tasks: [generateTask({ name: "task1", id: "task1" })],
+          }),
+          generateStep({
+            name: "step2",
+            step_number: 2,
+            section: "START",
+            tasks: [generateTask({ name: "task2", id: "task2" })],
+          }),
+        ],
+      });
+    });
 
-    const subject = renderMiniRoadMap("task3");
+    it("displays steps based on userData preferences", async () => {
+      const subject = renderMiniRoadMap("task1", userData);
+      expect(subject.getByText("task2")).toBeInTheDocument();
+    });
 
-    const sectionStart = subject.getByTestId("section-start");
-    const sectionPlan = subject.getByTestId("section-plan");
+    it("does not displays steps if not in userData preferences", async () => {
+      const subject = renderMiniRoadMap("task2", userData);
+      expect(subject.queryByText("task1")).not.toBeInTheDocument();
+    });
 
-    expect(within(sectionStart).getByText("step2")).toBeVisible();
-    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+    it("updates userData preferences on active step", async () => {
+      const subject = renderMiniRoadMap("task1", userData);
+      expect(subject.getByText("task1")).toBeInTheDocument();
+      await waitFor(() => expect(currentUserData().preferences.roadmapOpenSteps.sort()).toEqual([1, 2]));
+    });
+
+    it("updates userData on step click", async () => {
+      const subject = renderMiniRoadMap("task2", userData);
+      expect(subject.queryByText("task1")).not.toBeInTheDocument();
+      fireEvent.click(subject.getByText("step1"));
+      expect(subject.getByText("task1")).toBeInTheDocument();
+      expect(subject.getByText("task2")).toBeInTheDocument();
+      await waitFor(() => expect(currentUserData().preferences.roadmapOpenSteps.sort()).toEqual([1, 2]));
+    });
+    it("updates userData on active step click ", async () => {
```

Comment:
Thinking of the test as documentation, I'm looking at these test names and trying to figure out why "step click" and "active step click" would be different. But I see from the expectations (I think) that the difference is that an active step click CLOSES (removes from array) that step, whereas otherwise it OPENS (adds to array)

maybe we could make that clearer in test names to distinguish why the tests exist? for example `it('adds step to userData openSteps when non-active step is clicked")` and so on